### ISSUE TITLE
uv tool install does not respect index-url from pyproject.toml

### ISSUE THREAD

**Issue by benjs** [astral-sh/uv#8164]
UV tool does not look into the pyproject.toml for already specified indices.

pyproject.toml:
```toml
[tool.uv]
index-url = "https://<my-index-url.com>/pypi/pypi/simple"
```

command:
```bash
uv tool install ruff
```

uv --version:
```
uv 0.4.18
```

Specifying the index url with `--index-url` worked, but is annoying :)
Thanks for this great tool!

**Comment by charliermarsh** [comment]
Thanks! This is actually intentional right now. We have a separate design for "project-level tools" (i.e., tools that would be versioned and installed as part of the project, but given their own independent virtual environments) that we want to implement to solve this specific use-case. It's part of #3560.

**Comment by charliermarsh** [comment]
I'm gonna close in favor of that issue since we're tracking it centrally as part of that effort.

**Comment by KyzEver** [comment]
I'm also trying to configure `uv tool install` to adhere to a configuration provided in the pyproject toml and not rely on the additional `--default-index` flag, preferably.

This issue was closed in favor of #3560, but that issue has also been resolved & closed without any mention of tool installation index. I would like this to be a feature of uv, if possible.

Is it acceptable to reopen this issue?

**Comment by codethief** [comment]
@KyzEver I'd suggest we rather rally behind one of the open tickets, see [my comment here](https://github.com/astral-sh/uv/issues/3560#issuecomment-2902752225).
